src/main/java/com/google/firebase/projectmanagement/FirebaseProjectManagementServiceImpl.java

```diff
@@ -0,0 +1,909 @@
+      @Override
+      protected List<T> execute() throws FirebaseProjectManagementException {
+        String url = String.format(
+            "%s/v1beta1/projects/%s/%s?page_size=%d",
+            FIREBASE_PROJECT_MANAGEMENT_URL,
+            projectId,
+            platformResourceName,
+            MAXIMUM_LIST_APPS_PAGE_SIZE);
+        ImmutableList.Builder<T> builder = ImmutableList.<T>builder();
+        ListAppsResponse parsedResponse;
+        do {
+          parsedResponse = new ListAppsResponse();
+          makeGetRequest(url, parsedResponse, projectId, "Project ID");
+          if (parsedResponse.apps == null) {
+            break;
+          }
+          for (AppResponse app : parsedResponse.apps) {
+            builder.add(createAppFromAppIdFunction.apply(app.appId));
+          }
+          url = String.format(
+              "%s/v1beta1/projects/%s/%s?page_token=%s&page_size=%d",
+              FIREBASE_PROJECT_MANAGEMENT_URL,
+              projectId,
+              platformResourceName,
+              parsedResponse.nextPageToken,
+              MAXIMUM_LIST_APPS_PAGE_SIZE);
+        } while (!Strings.isNullOrEmpty(parsedResponse.nextPageToken));
+        return builder.build();
+      }
+    };
+  }
+
+  private static class ListAppsResponse {
+    @Key("apps")
+    private List<AppResponse> apps;
+
+    @Key("nextPageToken")
+    private String nextPageToken;
+  }
+
+  /* createAndroidApp, createIosApp */
+
+  @Override
+  public AndroidApp createAndroidApp(String projectId, String packageName, String displayName)
+      throws FirebaseProjectManagementException {
+    return blockOnFutureAndHandleExceptions(
+        createAndroidAppAsync(projectId, packageName, displayName),
+        packageName,
+        "Package name",
+        "Unable to create Android App");
+  }
+
+  @Override
+  public ApiFuture<AndroidApp> createAndroidAppAsync(
+      String projectId, String packageName, String displayName) {
+    checkArgument(!Strings.isNullOrEmpty(packageName), "package name must not be null or empty");
+    return
+        transform(
+            transformAsync(
+                createAndroidAppOp(projectId, packageName, displayName).callAsync(app),
+                new WaitOperationFunction(projectId),
+                scheduledExecutor),
+            createAndroidAppFromAppIdFunction,
+            scheduledExecutor);
+  }
+
+  @Override
+  public IosApp createIosApp(String projectId, String bundleId, String displayName)
+      throws FirebaseProjectManagementException {
+    return blockOnFutureAndHandleExceptions(
+        createIosAppAsync(projectId, bundleId, displayName),
+        bundleId,
+        "Bundle ID",
+        "Unable to create iOS App");
+  }
+
+  @Override
+  public ApiFuture<IosApp> createIosAppAsync(
+      String projectId, String bundleId, String displayName) {
+    checkArgument(!Strings.isNullOrEmpty(bundleId), "bundle ID must not be null or empty");
+    return
+        transform(
```

Comment:
Please qualify these as `ApiFutures.transform()`. In general all static references should be qualified. Guava preconditions are the only acceptable exception.